ocs_ci/ocs/cluster.py

```diff
@@ -619,3 +619,35 @@ def validate_pdb_creation():
         assert required == given, f"{required} was not created"
 
     logger.info(f"All required PDBs created: {pdb_required}")
+
+
+def get_node_resource_utilization(nodename=None):
+    """
+    Gets the node's cpu and memory utilization in percentage
+
+    Args:
+        nodename (str): The node name
+
+    Returns:
+        cpu_utilization (int) : if nodename is present
```

Comment:
I don't really like the idea of the return type changing based on arguments provided. I think it would be better to keep the return type and format consistent regardless of if `nodename` is provided. This way the user can call the function and handle the return value in the same way for both cases.

What if we instead return a dict that includes data for each node? Where if the `nodename` is provided the dict length will just be 1.

Something like this:
```python
resource_utilization_data = {
    'example-node-1': {
        'cpu': <cpu_utilization>,
        'memory': <memory_utilization>
    },
    'example-node-2': {
        'cpu': <cpu_utilization>,
        'memory': <memory_utilization>
    }
}
```
What do you think?

Reply:
ACK, I will send in next commit

Reply:
@coady Thanks for the pointer, Modified